substrate/frame/session/src/migrations/v1.rs

```diff
@@ -15,182 +15,75 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use core::str;
-use sp_io::hashing::twox_128;
-
+use crate::{Config, DisabledValidators as NewDisabledValidators, Pallet, Vec};
 use frame_support::{
-	storage::{generator::StorageValue, StoragePrefixedMap},
-	traits::{
-		Get, GetStorageVersion, PalletInfoAccess, StorageVersion,
-		STORAGE_VERSION_STORAGE_KEY_POSTFIX,
-	},
-	weights::Weight,
+	pallet_prelude::{Get, ValueQuery, Weight},
+	traits::UncheckedOnRuntimeUpgrade,
 };
+use sp_staking::offence::OffenceSeverity;
 
-use crate::historical as pallet_session_historical;
+#[cfg(feature = "try-runtime")]
+use sp_runtime::TryRuntimeError;
 
-const LOG_TARGET: &str = "runtime::session_historical";
+#[cfg(feature = "try-runtime")]
+use frame_support::ensure;
+use frame_support::migrations::VersionedMigration;
 
-const OLD_PREFIX: &str = "Session";
+/// This is the storage getting migrated.
+#[frame_support::storage_alias]
+type DisabledValidators<T: Config> = StorageValue<Pallet<T>, Vec<u32>, ValueQuery>;
 
-/// Migrate the entire storage of this pallet to a new prefix.
-///
-/// This new prefix must be the same as the one set in construct_runtime.
-///
-/// The migration will look into the storage version in order not to trigger a migration on an up
-/// to date storage. Thus the on chain storage version must be less than 1 in order to trigger the
-/// migration.
-pub fn migrate<T: pallet_session_historical::Config, P: GetStorageVersion + PalletInfoAccess>(
-) -> Weight {
-	let new_pallet_name = <P as PalletInfoAccess>::name();
+pub trait MigrateDisabledValidators {
```

Comment:
This is implemented in Staking and passed to the migration struct. Technically could have just migrated the storage from older prefix (staking) to newer but this seemed cleaner.